pkg/providers/vsphere/vsphere.go

```diff
@@ -946,16 +956,16 @@ func (p *vsphereProvider) GenerateCAPISpecForCreate(ctx context.Context, _ *type
 	return controlPlaneSpec, workersSpec, nil
 }
 
-func (p *vsphereProvider) createSecret(ctx context.Context, cluster *types.Cluster, contents *bytes.Buffer) error {
+func (p *vsphereProvider) updateAllVsphereSecrets(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec, contents *bytes.Buffer) error {
```

Comment:
this method uses 3 different patterns for k8s resource generation: go templates, go string formatting and go struct yaml marshaling.

Can we just pick one and use it for all secrets? I'm in favor of only using go struct marshaling.